components/BlurryPoster/BlurryPoster.tsx

```diff
@@ -95,6 +109,53 @@ export const BlurryPoster = ({
   } = formatDate(publishedDate);
 
   let i = 1;
+  /**
+   * Foreground image/video
+   */
+  const hasVideo = !!videoWebm || !!videoMp4;
+  const hasMedia = !!imageSrc || hasVideo;
+
+  /**
+   * Background image/video
+   */
+  const hasBgVideo = !!bgVideoWebm || !!bgVideoMp4;
+  const bgVideoRef = useRef<HTMLVideoElement>(null);
+  const isBgVideoInView = useInView(bgVideoRef, { once: false, amount: 0.1 });
+  const [isBgPlaying, setIsBgPlaying] = useState<boolean>(false);
+  const [isBgUserPaused, setIsBgUserPaused] = useState<boolean>(false);
+
+  // Toggle foreground video play/pause
+  const toggleBgVideo = () => {
+    if (!bgVideoRef.current) return;
+
+    setIsBgPlaying((prev) => {
+      if (prev) {
+        bgVideoRef.current?.pause();
+        setIsBgUserPaused(true);
+      } else {
+        bgVideoRef.current
+          ?.play()
+          .catch(() => {});
```

Comment:
Tell me more about this catch.

Reply:
I added this because sometimes video playback might be blocked or it can't be played for some reason, which can cause issues with DOM errors. This catches the error silently and prevents that from causing issue - I could have it display a console error or warning as well.

Reply:
Honestly I can certainly do more with error handling, but error should be rare. So in the interest of time, I'll come back to this later @sherakama 😂

Reply:
I'll let you judge the time/effort to spend here but overall I am not a fan of squashing errors quietly as they tend to be very difficult to troubleshoot. I'd rather see the error logged.

Reply:
I can update for it to show a console warning 👍🏼